spec/app/domain/conjur/fetch_trusted_proxies_spec.rb

```diff
@@ -34,17 +121,177 @@
   #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
 
   context "A valid trusted proxies configuration" do
+    context "trusted proxies contains an empty list of IPs addresses" do
+      context "when role.restricted_to returns empty array" do
+        subject do
+          Conjur::FetchTrustedProxies.new(
+              role_cls:                 mocked_role_class_empty_list,
+              validate_account_exists:  mocked_account_validator,
+              ).call(
+              account: valid_account
+          )
+        end
+
+        it "returns an empty list" do
+          expect( subject ).to eq(empty_trusted_proxies_list)
+        end
+      end
+
+      context "when role.restricted_to returns nil" do
+        subject do
+          Conjur::FetchTrustedProxies.new(
+              role_cls:                 mocked_role_class_return_nil,
+              validate_account_exists:  mocked_account_validator,
+              ).call(
+              account: valid_account
+          )
+        end
+
+        it "returns an empty list" do
+          expect( subject ).to eq(empty_trusted_proxies_list)
+        end
+      end
+    end
+
+    context "trusted proxies contains an list of IPs addresses without duplications" do
+      subject do
+        Conjur::FetchTrustedProxies.new(
+            role_cls:                 mocked_role_class_list_no_duplications,
+            validate_account_exists:  mocked_account_validator,
+            ).call(
+            account: valid_account
+        )
+      end
+
+      it "returns expected list" do
+        expect( subject ).to eq(trusted_proxies_list_no_duplications)
+      end
+    end
+
+    context "trusted proxies contains an list of IPs addresses with duplications" do
+      context "with 1 same IP address" do
+        subject do
+          Conjur::FetchTrustedProxies.new(
+              role_cls:                 mocked_role_class_list_duplications_1,
+              validate_account_exists:  mocked_account_validator,
+              ).call(
+              account: valid_account
+          )
+        end
+
+        it "returns list with no duplication" do
+          expect( subject ).to eq(expected_list_1)
+        end
+      end
+
+      context "with 3 same IP addresses" do
```

Comment:
There is 3 duplication ([ip_1, ip_2 , ip_3, ip_1, ip_2 , ip_3] ) and all over the test you wrote mocked_role_class_list_duplications_2, expected_list_2 , need to make it clear.